---
title: Calculating Basic Population Genetic Statistics from SNP Data
---

# Introduction

In this vignette, you will  calculate basic population genetic statistics from
SNP data using R packages. These statistics serve as exploratory analysis and
require to work at the population level. We will calculate: 

1. Genetic diversity,
2. Test Hardy Weinberg 
3. $F_{is}$ and global $F_{st}$.

The dataset used for these analyses are for lodgepole pine
(*Pinus contorta*, Pinaceae),  a plant species. More information on the dataset and
the species is available from [A. Eckert's website](http://eckertdata.blogspot.fr/). Here
we use the dataset with no presupposed idea of interpreting the results in a biological way.
To make the calculations faster, we work only with a subset of the full dataset.


# Load necessary resources and packages

```{r,packages, message=FALSE}
library("adegenet")
library("hierfstat")
library("pegas")
```

# Workflow for SNP data

## Import data

The data are stored in a text file (genotype=AA..). We will  import the dataset
into R as a data frame, and then  convert the SNP data file into a
"[genind](http://www.inside-r.org/packages/cran/adegenet/docs/.valid.genind)"
object.

The dataset ["Master_Pinus_data_genotype.txt" can be downloaded 
here](https://github.com/NESCent/popgenInfo/tree/master/data/Master_Pinus_data_genotype.txt).

The text file is a matrix of (550 rows x 3086 columns). It contains 4
extra columns: first column is the label of the individuals, the three other are
description of the region, all the other columns are for the genotypes as (AA or
AT...).

When you import the data into R, the data file needs to be in your working directory, 
or adjust the path in the `read.table()` invocation below accordingly.

```{r, data_import_df_show, eval = FALSE}
Mydata <- read.table("Master_Pinus_data_genotype.txt", header = TRUE)
dim(Mydata) # Matrix of dimension 550x3086
```

```{r, data_import_df_do, echo = FALSE}
Mydata <- read.table("../data/Master_Pinus_data_genotype.txt", header = TRUE)
dim(Mydata) # Matrix of dimension 550x3086
```

To work with the data, we need to convert the R object returned by `read.table()` to a "genind" object.
To achieve this, we create a matrix with only genotypes, and keep only a subset of the first 11
SNP loci (to make calculations faster). The result can then be converted to a "genind" object (for package *adegent*).
The "genind" object can then easily be converted into a "hierfstat" (package *hierfstat*) object.

```{r,data_conversion}
locus <- Mydata[, -c(1, 2, 3, 4, 17:3086)]    
colnames(locus) <- gsub("\\.", "_", colnames(locus)) # locus names can't have "."
ind <- as.character(Mydata$tree_id) # labels of the individuals
population <- as.character(Mydata$state) # labels of the populations
Mydata1 <- df2genind(locus, ploidy = 2, ind.names = ind, pop = population, sep = "")
Mydata1
nAll(Mydata1) # Number of alleles per locus
Mydata2 <- genind2hierfstat(Mydata1) # Create hierfstat object
```

## Genetic diversity (observed and expected heterozygosity) 

With *adegenet*:

```{r, Genetic_Diversity}

div <- summary(Mydata1)
div
names(div)

plot(div$Hobs, xlab="Loci number", ylab="Observed Heterozygosity", 
	 main="Observed heterozygosity per locus")

plot(div$Hobs,div$Hexp, xlab="Hobs", ylab="Hexp", 
	 main="Expected heterozygosity as a function of observed heterozygosity per locus")

bartlett.test(list(div$Hexp, div$Hobs)) # a test : H0: Hexp = Hobs
```
We observed that heterozygosity varies among loci.
We observed no difference between expected and observed heterozygosity.


Basic statistics with *hierfstat*. Populations are states. The function `basic.stats()`
provides the observed heterozygosity ($H_o$), mean gene diversities within
population ($H_s$), $F_{is}$, and $F_{st}$. The function `boot.ppfis()` provides
confidence interval for $F_{is}$. The function `indpca()` does an PCA on the
centered matrix of individuals' allele frequencies.

```{r, basic_statistics}
basicstat <- basic.stats(Mydata2, diploid = TRUE, digits = 2) 
names(basicstat)   
boot.ppfis(Mydata2) 
x <- indpca(Mydata2) 
plot(x, cex = 0.7)

```

## Testing for Hardy-Weinberg Equilibrium  

We used the function `hw.test()` from the *pegas* package.

```{r, Hardy-Weinberg_test}

hw.test(Mydata1, B = 1000)
```

We get for each locus a test of significance of the null hypothesis: $H_0$ the
locus is in HW equilibrium in the population. All but one locus are in HW
equilibrium.



# Conclusion

## What did we learn today?

In this vignette, we learned how to explore the patterns of genetic diversity
and how to estimate the level of genetic differentiation in one population.
Also, you have an idea of potential violations of the dataset to the null
Wright-Fischer model.

## What is next?

You may now want to move into looking into population differentiation in more
detail (See [Calculating genetic differentiation and clustering methods from SNP
data](DifferentiationSNP.html))

# References

Eckert, A. J., A. D. Bower, S. C. González-Martínez, J. L. Wegrzyn, G. Coop and
D. B. Neale. 2010. Back to nature: Ecological genomics of loblolly pine (Pinus taeda, Pinaceae).
Molecular Ecology 19: 3789-3805. doi:[10.1111/j.1365-294X.2010.04698.x](http://dx.doi.org/10.1111/j.1365-294X.2010.04698.x)


# Contributors

- Stéphanie Manel (Author)
- Zhian Kamvar (edits)

# Session Information

This shows us useful information for reproducibility. Of particular importance are the versions of R and the packages used to create this workflow. It is considered good practice to record this information with every analysis.

```{r, sessioninfo}
options(width = 100)
devtools::session_info()
```
